scripts/fabric_links_monitoring.py

```diff
@@ -0,0 +1,183 @@
+        for namespace in self.namespaces:
+            self.appdb[namespace] = self.appl_db(namespace)
+            self.configdb[namespace] = self.config_db(namespace)
+            self.statedb[namespace] = self.state_db(namespace)
+            self.link_monitor[namespace] = FabricLinkErrMontoring(
+                namespace=namespace, port_name_map='COUNTERS_FABRIC_PORT_NAME_MAP',
+                filter_for_up_ports=self.filter_for_up_ports)
+        self.error_counter_names = ['SAI_PORT_STAT_IF_IN_ERRORS']
+        self.threshold = 60
+
+    @property
+    def namespaces(self):
+        namespaces = []
+        SonicDBConfig.load_sonic_global_db_config()
+        namespaces = multi_asic.get_namespace_list()
+        return namespaces
+        
+    def config_db(self,namespace):
+        '''Returns the config db object for the namespace'''
+        db = SonicV2Connector(namespace=namespace)
+        db.connect('CONFIG_DB')
+        return db
+    
+    def appl_db(self, namespace):
+        '''Returns the application db object for the namespace'''
+        db = SonicV2Connector(namespace=namespace)
+        db.connect('APPL_DB')
+        return db
+
+    def state_db(self, namespace):
+        '''Returns the state db object for the namespace'''
+        db = SonicV2Connector(namespace=namespace)
+        db.connect('STATE_DB')
+        return db
+
+    def get_port_status(self, namespace, port_name):
+        ''' Returns operational status of give port 
+            When port is admin shut in CONFIG_DB, it might take some time for port to be oper down in APPL_DB.
+            As such, status of ports which are admin shut in config DB is reported as down
+        '''
+        port_info = self.statedb[namespace].get_all('STATE_DB', 'FABRIC_PORT_TABLE|{}'.format(port_name))
+        return port_info['STATUS']
+
+    def filter_for_up_ports(self, namespace, port_map):
+        ''' Returns only internal (backend) ports which are operationally up for monitoring '''
+        filtered_port_map = {}
+        if port_map:
+            for k,v in port_map.items():
+                if self.get_port_status(namespace, k) == 'up':
+                    filtered_port_map[k] = v
+        return filtered_port_map
+
+    def monitor(self):
+        '''
+        Monitors the error counters for each fabric port in each namespace.
+        If any error counters are above the specified threshold, a syslog message is generated.
+        '''
+        all_error_ports = []
+        while True:
+            for namespace, link_mon in self.link_monitor.items():
+                err_port_per_ns = link_mon.monitor(namespace, self.error_counter_names, self.threshold)
+                if err_port_per_ns:
+                    syslog.syslog(syslog.LOG_CRIT,
+                                '{} fabric ports in {} have errors above threshold'.format(len(err_port_per_ns), namespace))
+            time.sleep(60)
+
+def main():
+    if multi_asic.is_multi_asic():
```

Comment:
do we need to add check to run this script only on voq chassis?

Reply:
Yes.